core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -454,26 +465,33 @@ class ReplicaManager(val config: KafkaConfig,
   }
 
   def getPartitionOrException(topicPartition: TopicPartition, expectLeader: Boolean): Partition = {
+    getPartitionOrError(topicPartition, expectLeader) match {
+      case Left(error) => throw error.exception(s"Failed to get Partition $topicPartition")
```

Comment:
This message is a bit generic compared to what we were getting before.  Maybe that's fine since the error code/exception are mostly self-explanatory (`KAFKA_STORAGE_ERROR` is not as obvious as the others, but maybe close enough). Maybe we can tweak the generic error though. A KafkaStorageException with message "Failed to get Partition $topicPartition" may be a bit confusing. I don't have great ideas though.

Reply:
I added a special case for the storage error and tried to improve the general message.